version: "3"

services:
  namenode:
    build: ./Namenode
    hostname: 10.2.14.251
    container_name: namenode
    volumes:
      - hadoop_namenode:/hadoop/dfs/name
      - ./data/:/hadoop-data/input
      - ./map_reduce/:/hadoop-data/map_reduce
    environment:
      - CLUSTER_NAME=test
    ports:
      - "9870:9870"
      - "8020:8020"
    networks:
      - hadoop_network

  resourcemanager:
    build: ./Resourcemanager
    container_name: resourcemanager
    hostname: resourcemanager
    restart: on-failure
    depends_on:
      - namenode
      - datanode1
    ports:
      - "8089:8088"
    networks:
      - hadoop_network

  nodemanager1:
    build: ./Nodemanager
    hostname: nodemanager1
    container_name: nodemanager1
    depends_on:
      - namenode
      - datanode1
    ports:
      - "8042:8042"
    networks:
      - hadoop_network


  datanode1:
    build: ./Datanode
    hostname: 10.2.14.251
    depends_on:
      - namenode
    ports:
      - "9864:9864"
    volumes:
      - hadoop_datanode1:/hadoop/dfs/data
    networks:
      - hadoop_network


volumes:
  hadoop_namenode:
  hadoop_datanode1:


networks:
  hadoop_network:
    name: hadoop_network
